frontend/epfl-caption-cards/controller.php

```diff
@@ -32,7 +32,11 @@ function epfl_caption_cards_block( $attributes ) {
             <?php for($i = 1; $i <= 10; $i++): ?>
             <?php if (!empty($attributes['title'.$i])) : ?>
                 <div class="col-sm-6 col-xl-4">
+                    <?php if ($link): ?>
```

Comment:
Comme je ne sais pas d'où vient la variable $link, je propose ceci plutôt, dans le respect dans la boucle en cours :
```suggestion
                    <?php if (!empty($attributes['link'.$i])): ?>
```